astropy/units/format/generic.py

```diff
@@ -645,7 +616,29 @@ def _format_unit_list(cls, units):
 
     @classmethod
     def to_string(cls, unit):
-        return _to_string(cls, unit)
+        parts = []
+
+        if unit.scale != 1:
+            parts.append(f"{unit.scale:g}")
+
+        if len(unit.bases):
+            nominator, denominator = utils.get_grouped_by_powers(
+                unit.bases, unit.powers
+            )
+            if len(nominator):
+                parts.append(cls._format_unit_list(nominator))
```

Comment:
I don't understand why some of the class methods are public (`cls.format_exponential_notation()`) and others are marked as private (`cls._format_unit_list()`). Unless there is a specific reason here, shouldn't the `format` functions all have the same visibility?

Reply:
This is well before my time, but I think the idea was that the public methods are the only ones also used outside of the class - e.g., `format_exponential_notation` is not used only for the scale, but also by `Quantity` to format the values.

Maybe something to reconsider, though! (Perhaps best after finishing the refactor.)